### PR TITLE
Feat/my actions

### PR DESCRIPTION
**Pull request by 0xRowdy**
Created mobile and desktop ui for My Actions page.
Issue #14 

Desktop
Screenshot 2024-02-13 at 8 21 01 PM

Mobile
Screenshot 2024-02-13 at 8 21 32 PM
Screenshot 2024-02-13 at 8 21 45 PM
Screenshot 2024-02-13 at 8 21 50 PM

### CODE REVIEW

**Comment by 0xRowdy** [comment]
I'm realizing that the folder components/actions might not be clear. 
We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

File: app/components/actions/history-card.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import {
+	Card,
+	CardContent,
+	CardDescription,
+	CardHeader,
+	CardTitle,
+} from "~/components/ui/card";
+import { Separator } from "~/components/ui/separator";
+import { formatCurrency, formatDate } from "~/lib/utils";
+import DynamicCategoryIcon from "../ui/dynamic-category-icon";
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:13]:
```suggestion
import DynamicCategoryIcon from "~/components/ui/dynamic-category-icon";
```

File: app/components/actions/history-card.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { Badge } from "~/components/ui/badge";
+import {
+	Card,
+	CardContent,
+	CardDescription,
+	CardHeader,
+	CardTitle,
+} from "~/components/ui/card";
+import { Separator } from "~/components/ui/separator";
+import { formatCurrency, formatDate } from "~/lib/utils";
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:12]: move lib imports above components

File: app/components/actions/history-card.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+				</div>
+				<div className="flex flex-col gap-2 md:flex-row md:items-center">
+					<img
+						src={img.src}
+						alt={img.alt}
+						className="w-30 md:w-auto h-full md:h-auto overflow-clip object-cover object-center rounded-3xl mb-2"
+					/>
+					<div className="flex flex-col gap-3">
+						<CardTitle>{title}</CardTitle>
+						{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:47]:
```suggestion
```

File: app/components/actions/history-card.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+						<div className="flex gap-2 pt-2">
+							<Badge
+								variant="default"
+								className="rounded-3xl justify-between items-center"
+							>
+								<DynamicCategoryIcon category={category} />
+								{/* <Salad className="text-vd-orange-400 stroke-1 size-4 mr-1" /> */}
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:54]:
```suggestion
```

File: app/components/actions/history-card.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+								<p className="font-light text-xs ml-1">{category}</p>
+							</Badge>
+							{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:57]:
```suggestion
```

File: app/components/actions/history-card.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+							<Badge
+								variant="default"
+								className="rounded-3xl justify-between items-center"
+							>
+								<DynamicCategoryIcon category="Location" />
+								{/* <MapPin className="text-vd-orange-400 stroke-1 size-4 mr-1" /> */}
```
Comment (0xRowdy) [inline app/components/actions/history-card.tsx:63]:
```suggestion
```

File: app/components/actions/summary.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+			<Card className="rounded-3xl">
+				<CardHeader>
+					<CardTitle className="text-sm font-normal">
+						Issues I care about:
+					</CardTitle>
+				</CardHeader>
+				<CardContent>
+					{/* TODO: Make Dynamic based on user */}
+					<div className="grid grid-cols-1 md:grid-cols-2 gap-2 md:gap-4">
+						{/* TODO: Refactor to use DynamicCategoryIcon component from Beyonder */}
```
Comment (0xRowdy) [inline app/components/actions/summary.tsx:39]:
```suggestion
```

**Review by CJ-Rose — CHANGES_REQUESTED** [review]

**Comment by CJ-Rose** [comment]
> I'm realizing that the folder components/actions might not be clear.
We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

i find the the current structure to be clear. can also have `components/reports` and `components/report-detail` folders to hold the components used in those routes.
removing the folder and renaming files `actions-history`, `reports-voicedeck-stats`, etc could be good too, as the files would appear grouped together alphabetically.
it makes more sense to me to keep all the components inside the components folder (along with `global` and `ui`) rather than splitting these off into the related route folders.

File: app/routes/_index.tsx [hunk 8]
```diff
@@ -36,7 +36,7 @@ import { siteConfig } from "~/config/site";
 import { Report } from "~/types";
 import { fetchReports } from "../impact-reports.server";
 
-const iconComponents: { [key: string]: LucideIcon } = {
+export const iconComponents: { [key: string]: LucideIcon } = {
```
[the remark this answers is not in the document]
Comment (0xRowdy) [reply app/routes/_index.tsx:39]: Done

File: app/components/actions/summary.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import { Badge } from "~/components/ui/badge";
+import { Card, CardContent, CardHeader, CardTitle } from "~/components/ui/card";
+import { iconComponents } from "~/routes/_index";
```
[the remark this answers is not in the document]
Comment (0xRowdy) [reply app/components/actions/summary.tsx:3]: Good catch! Done

**Comment by thebeyondr** [comment]
> We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.

 `component/my-actions` might be clearer. 
 
 However, there are even more possibilities with Remix. Let's examine one: **Colocation**
 
 ```
 routes/
├── [...]
├── my-actions/
│   ├── index.tsx # /my-actions 
│   ├── history-card.tsx
│   ├── summary.tsx
```

File: app/lib/utils.ts [hunk 10]
```diff
@@ -4,3 +4,21 @@ import { twMerge } from "tailwind-merge";
 export function cn(...inputs: ClassValue[]) {
 	return twMerge(clsx(inputs));
 }
+
+export function formatDate(date: Date) {
```
Comment (thebeyondr) [inline app/lib/utils.ts:8]: noice

**Review by thebeyondr — CHANGES_REQUESTED** [review]
Looking good so far, but I can't approve yet because: 

- [x] small css typo (600 uses 400 color). 
- [ ] Empty state unaccounted for
- [x] Some cards have wrong bg color (see figma)
- [ ] Some bento areas and cards have shadows where they should not (see figma)

**Comment by 0xRowdy** [comment]
> > We could address this in a few ways. Have a components folder in the folder of the route, or I could rename the components to indicate the page they are associated with ex. actions-history, actions-history-card, etc. Open to other ideas.
> 
> `components/my-actions` might be clearer.
> 
> However, there are even more possibilities with Remix. Let's examine one: **Colocation**
> 
> ```
> routes/
> ├── [...]
> ├── my-actions/
> │   ├── index.tsx # /my-actions 
> │   ├── history-card.tsx
> │   ├── summary.tsx
> ```

Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.

**Comment by thebeyondr** [comment]
> Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.

Here's the [direct reference in Remix v2](https://remix.run/docs/en/main/file-conventions/routes#folders-for-organization). I got some naming things wrong. If the user is unauthenticated then they wouldn't have a `My Actions` link. and if they try to access the route they should be redirected to '/'

**Comment by 0xRowdy** [comment]
> > Yeah, I like this approach. Is that a good home for components or is that generally used for view/routes? I might need to do a little bit of re-factoring of this route due to the default state being the un-authenticated state.
> 
> Here's the [direct reference in Remix v2](https://remix.run/docs/en/main/file-conventions/routes#folders-for-organization). I got some naming things wrong.
> 
> If the user is unauthenticated then they wouldn't have a `My Actions` link in nav. If they try to access the route manually they should be redirected to '/'

Yeah that is what I would expect. Currently, the design looks like it would ask for auth if you try to navigate to my-actions. We could add connect wallet to the header and not show the My Actions Link unless the user has signed in with their wallet. That might kinda render the design useless unless we did some route handling and made the my-actions unauthenticated state its own route called auth, log-in or sign-in.

We've got some options here.

File: tailwind.config.js [hunk 11]
```diff
@@ -19,9 +19,10 @@ module.exports = {
     extend: {
       colors: {
         "vd-beige": {
-          200: "rgb(var(--vd-beige-200) / <alpha-value>)",
+          600: "rgb(var(--vd-beige-400) / <alpha-value>)",
```
[the remark this answers is not in the document]
Comment (0xRowdy) [reply tailwind.config.js:22]: done

**Comment by 0xRowdy** [comment]
> Looking good so far, but I can't approve yet because:
> 
> * [ ]  small css typo (600 uses 400 color).
> * [ ]  Empty state unaccounted for
> * [x]  Some cards have wrong bg color (see figma)
> * [x]  Some bento areas and cards have shadows where they should not (see figma)

Can you help me understand?

My Contribution card = --vd-blue-200: 194 214 223;

Screenshot 2024-02-15 at 5 20 49 PM

#of Reports card = --vd-beige-300: 241 224 215;
Screenshot 2024-02-15 at 5 40 55 PM


> Looking good so far, but I can't approve yet because:
> 
> * [ ]  small css typo (600 uses 400 color).
> * [ ]  Empty state unaccounted for
> * [ ]  Some cards have wrong bg color (see figma)
> * [ ]  Some bento areas and cards have shadows where they should not (see figma)

I don't follow you on Empty state unaccounted for. if you are referring to history it has an empty state when !history. I originally added a check for the history object like {history && history.map => ... but the linter complained and had me update it to {history?.map

Lmk if this is what you were talking about with empty state or something else.

**Comment by thebeyondr** [comment]
> Lmk if this is what you were talking about with empty state or something else.

Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

**Comment by 0xRowdy** [comment]
> > Lmk if this is what you were talking about with empty state or something else.
> 
> Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

I generae

> > Lmk if this is what you were talking about with empty state or something else.
> 
> Yes I meant History empty state. Seems I missed it my bad. I refactored it. Not sure why you're not using default exports for components though but I changed History to be one.

My general rule of thumb is to use default exports for pages and named exports for components. I've had several projects and repos that enforce only named exports. It used to be more of an issue with tree shaking and with IDEs offering better autocomplete and import suggestions with named exports.

**Comment by thebeyondr** [comment]
> My general rule of thumb is to use default exports for pages and named exports for components. I've had several projects and repos that enforce only named exports. It used to be more of an issue with tree shaking and with IDEs offering better autocomplete and import suggestions with named exports.

makes sense, thanks for the explaining 🫡

**Review by CJ-Rose — APPROVED** [review]

**Comment by 0xRowdy** [comment]
@thebeyondr Anything else here? I got all the requested changes but you still might need to re-approve.

**Review by thebeyondr — APPROVED** [review]

**Pull request merged**